File: src/main/java/ArraysHash/E3TwoSum.java
package ArraysHash;

import java.util.HashMap;

public class E3TwoSum {


    public int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if ((nums[i] + nums[j]) == target) {
                    return new int[]{i, j};
                }
            }
        }
        return new int[]{};
    }


    public int[] twoSum2(int[] nums, int target) {
        // valor - index
        HashMap<Integer, Integer> hs = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int resta = target - nums[i];
            if (!hs.containsKey(resta)) {
                hs.put(nums[i], i);
            } else {
                return new int[]{hs.get(resta), i};
            }
        }
        return new int[]{};
    }

    public static void main(String[] args) {
        int[] arreglo = new int[]{2, 7, 11, 15};
        E3TwoSum o = new E3TwoSum();
        o.twoSum(arreglo, 9);
    }
}

File: src/main/kotlin/E3TwoSum.kt
fun twoSum(nums: IntArray, target: Int): IntArray? {
    for (i in nums.indices) {
        for (j in i + 1 until nums.size) {
            if ((nums[i] + nums[j]) == target) {
                return intArrayOf(i, j)
            }
        }
    }
    return intArrayOf()
}

fun twoSum2(nums: IntArray, target: Int): IntArray? {
    val hs = HashMap<Int, Int>()
    for (i in nums.indices) {
        val resta = target - nums[i]
        if (!hs.containsKey(resta)) {
            hs[nums[i]] = i
        } else {
            return intArrayOf(hs[resta]!!, i)
        }
    }
    return intArrayOf()
}


fun main() {

    val numeros = intArrayOf(2, 7, 11, 15)
    twoSum(numeros, 9)

}


